src/main/java/org/jenkinsci/plugins/github/extension/GHEventsSubscriber.java

```diff
@@ -55,12 +55,13 @@
      * This method called when root action receives webhook from GH and this extension is interested in such
      * events (provided by {@link #events()} method). By default do nothing and can be overrided to implement any
      * parse logic
-     * Don't call it directly, use {@link #processEvent(GHEvent, String)} static function
+     * Don't call it directly, use {@link #processEvent(GHEvent, String, String)} static function
```

Comment:
More flexibility in terms of security. Only having the ability to use a global secret means more people would potentially have to share the key among themselves if setting up the hook manually. Having the option to use a project-specific key (that would also override any global key) mitigates this issues by letting them specify their own key.